dali/operators/generic/one_hot.cu

```diff
@@ -0,0 +1,96 @@
+// Copyright (c) 2019, NVIDIA CORPORATION. All rights reserved.
+//
+// Licensed under the Apache License, Version 2.0 (the "License");
+// you may not use this file except in compliance with the License.
+// You may obtain a copy of the License at
+//
+//     http://www.apache.org/licenses/LICENSE-2.0
+//
+// Unless required by applicable law or agreed to in writing, software
+// distributed under the License is distributed on an "AS IS" BASIS,
+// WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+// See the License for the specific language governing permissions and
+// limitations under the License.
+
+#include <vector>
+#include "dali/operators/generic/one_hot.h"
+#include "dali/operators/generic/one_hot.cuh"
+
+namespace dali {
+
+class OneHotGPU : public OneHot<GPUBackend> {
+ public:
+  explicit OneHotGPU(const OpSpec &spec) : OneHot<GPUBackend>(spec) {
+    int64_t samples_size = batch_size_ * sizeof(detail::SampleDesc);
```

Comment:
Please infer the batch size from the input, and move that to setup_impl

Reply:
done!